### PR TITLE
Simplify af- & r_anal_fcn_del

### PR DESCRIPTION
**Pull request by MaskRay**
r_anal_fcn_del should not treat UT64_MAX differently.

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -2170,8 +2170,8 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 		break;
 	case '-': // "af-"
 		if (!input[2] || !strcmp (input + 2, "*")) {
-			r_anal_fcn_del_locs (core->anal, UT64_MAX);
-			r_anal_fcn_del (core->anal, UT64_MAX);
+			r_list_purge (core->anal->fcns);
```
Comment (radare) [inline libr/core/cmd_anal.c]: what about having r_anal_fcn_del_all () instead of accessing internal fields from here?

Comment (MaskRay) [reply libr/core/cmd_anal.c]: Do this when there is another command deleting all functions.

Comment (MaskRay) [reply libr/core/cmd_anal.c]: This  also fixes an issue when you craft a function with `af+` which `r_anal_fcn_in` will return false.

Comment (radare) [reply libr/core/cmd_anal.c]: af-* deletes all functions

Comment (radare) [reply libr/core/cmd_anal.c]: just move this list_purge into that anal function

Comment (MaskRay) [reply libr/core/cmd_anal.c]: > just move this list_purge into that anal function

?

**Comment by radare** [comment]
will fix myself is easier to explain in C

**Comment by radare** [comment]
but maybe is better to just have a function that deletes all functions in a range, this way we can use it to delete all the functions if args are 0, -1, or be able to remove all the functions in a range

like this `R_API int r_anal_fcn_del_in_range(RAnal *a, ut64 from, ut64 to) {`

what do u think?

**Comment by MaskRay** [comment]
I agree. Extend it to `af- [from] [to]`?

* How to delete the current function?
* How to delete the functions in [0x100,0x200)?

And how about the renaming `afd` -> `af.` (if `af.` has not been used yet)

**Comment by radare** [comment]
af-$$

Or

af-.

**Comment by radare** [comment]
Agree on those changes

**Pull request merged**